src/controller/controller.cpp

```diff
@@ -41,11 +69,15 @@ void milk::controller_close()
     {
         std::cout << "Could not close gamecontroller" << SDL_GetError() << std::endl;
     }
+    
+    SDL_QuitSubSystem(SDL_INIT_HAPTIC);
+    haptic = NULL;
```

Comment:
Prefer nullptr to NULL - Bowl cut hair man.

Also you don't need the nullptr assignments. you only need that if you access that ptr later on in the block and risk accessing undefined memory